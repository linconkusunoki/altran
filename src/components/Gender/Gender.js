import React from 'react'
import PropTypes from 'prop-types'

const Gender = ({ color }) => {
  const male = ['Green', 'Red', 'Black']
  if (male.includes(color)) {
    return (
      <span role="img" aria-label="male">
        ðŸ‘¨
      </span>
    )
  }

  return (
    <span role="img" aria-label="female">
      ðŸ‘©
    </span>
  )
}

Gender.propTypes = {
  color: PropTypes.string.isRequired,
}

export default Gender
