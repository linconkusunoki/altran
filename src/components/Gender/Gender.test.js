import React from 'react'
import { render } from '@testing-library/react'
import Gender from './'

it('should render correctly', async () => {
  const { container, getByTestId } = render(<Gender color="Green" />)
  expect(container.firstChild).toMatchSnapshot()
  expect(getByTestId('male')).toBeTruthy()
})

it('should render a female', async () => {
  const { container, getByTestId } = render(<Gender color="Purple" />)
  expect(container.firstChild).toMatchSnapshot()
  expect(getByTestId('female')).toBeTruthy()
})
