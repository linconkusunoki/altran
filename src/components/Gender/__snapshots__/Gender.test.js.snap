// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a female 1`] = `
<span
  aria-label="female"
  data-testid="female"
  role="img"
>
  ðŸ‘©
</span>
`;

exports[`should render correctly 1`] = `
<span
  aria-label="male"
  data-testid="male"
  role="img"
>
  ðŸ‘¨
</span>
`;
